packages/shopping-cart/src/shopping-cart-manager.ts

```diff
@@ -60,10 +73,24 @@ function createShoppingCartManager(
 		const isStateChanged = newState !== state;
 		state = newState;
 
+		if ( state.cacheStatus === 'error' ) {
+			actionPromises.reject(
+				new ActionPromiseError(
+					state.loadingError ?? 'Unknown error',
+					state.loadingErrorType ?? 'Unknown type'
```

Comment:
Should we just copy the `loadingError` and `loadingErrorType` values without defaulting them? Then the types of `ActionPromiseError` fields are exactly the same as the `ShoppingCartManagerState`'s corresponding fields. We don't need to introduce confusing differences.

Reply:
In practice, if the `cacheStatus` is `'error'`, then both `loadingError` and `loadingErrorType` should always be set, but TypeScript knows that it's possible that somehow the state could be changed by future reducer additions and not follow that contract, so it complains here. These defaults, therefore, are here so that if they ever show up, we'll know that something is very wrong.

That said, it would be nice to avoid this and convince TypeScript that a `cacheStatus` of `'error'` _requires_ the other two variables to be set. I'm not sure if that's possible but I'll give it a try.

Reply:
I turns out that I _was_ able to convince TS about the rules, so I've removed these fallbacks!